modules/spark/src/main/scala/org/apache/spark/sql/ignite/IgniteExternalCatalog.scala

```diff
@@ -144,7 +161,9 @@ private[ignite] class IgniteExternalCatalog(defaultIgniteContext: IgniteContext)
 
             val ccfg = cache.getConfiguration(classOf[CacheConfiguration[Any, Any]])
 
-            ccfg.getQueryEntities.map(_.getTableName)
+            ccfg.getQueryEntities.flatMap(
+                e => Array(e.getTableName, normalizeSchemaName(ccfg.getName, ccfg.getSqlSchema)
```

Comment:
Let's add a test for a cache with special characters - `cache-for-table` or something similar.

Reply:
Looking at adding this test I realised there's an existing bug which means that table names with special characters don't work in the current codebase. I have a few changes to fix this but I need to go through most of the tests and change the SQL query assertions. Suggest I add this as part of a separate ticket otherwise this set of changes will balloon. I've created [IGNITE-9317](https://issues.apache.org/jira/browse/IGNITE-9317) for this purpose.

Reply:
> table names with special characters don't work in the current codebase.

Thanks for bringing this up.
Please, attach a reproducer to a ticket.
Or write a test in  pull request, if you decide to contribute solution.
Let's fix it!

Reply:
I'll fix it - I have assigned the ticket to myself.